fire/SFEquationsMod.F90

```diff
@@ -314,5 +331,139 @@ real(r8) function BackwardRateOfSpread(ros_front, wind_speed)
     end function BackwardRateOfSpread
 
     !-------------------------------------------------------------------------------------
+    
+    real(r8) function FireDuration(FDI)
+    !
+    !  DESCRIPTION:
+    !  Calculates fire duration [min]
+    !
+    !  Equation 14 in Thonicke et al. 2010
+    !
+    use SFParamsMod, only : SF_val_max_durat, SF_val_durat_slope
+
+    ! ARGUMENTS:
+    real(r8), intent(in) :: FDI  ! fire danger index [0-1]
+
+    FireDuration = (SF_val_max_durat + 1.0_r8)/(1.0_r8 + SF_val_max_durat*               &
+      exp(SF_val_durat_slope*FDI))
+
+  end function FireDuration
+
+  !---------------------------------------------------------------------------------------
+ 
+  real(r8) function LengthToBreadth(effective_windspeed, tree_fraction)
+  !
+  !  DESCRIPTION:
+  !  Calculates length to breadth ratio of fire ellipse [unitless], used for calculating area burnt 
+  !
+  !     Canadian Forest Fire Behavior Prediction System Ont.Int.Rep. ST-X-3, 1992
+  !     Information Report GLC-X-10, Wotten et al. 2009
+  !
+  use FatesConstantsMod, only : m_per_km, min_per_hr
+
+  ! ARGUMENTS:
+  real(r8), intent(in) :: effective_windspeed ! effective windspeed [m/min]
+  real(r8), intent(in) :: tree_fraction       ! tree fraction [0-1]
+  
+  ! LOCALS:
+  real(r8) :: windspeed_km_hr ! effective windspeed, converted to correct units [km/hr]
+  
+  ! CONSTANTS:
+  real(r8), parameter :: lb_threshold = 0.55_r8                ! tree canopy fraction below which to use grassland length-to-breadth eqn
+  real(r8), parameter :: m_per_min__to__km_per_hour = 0.06_r8  ! convert wind speed from m/min to km/hr
+
+  windspeed_km_hr = effective_windspeed/m_per_km*min_per_hr
+
+  if (windspeed_km_hr < 1.0_r8) then 
+    LengthToBreadth = 1.0_r8
+ else
+    if (tree_fraction > lb_threshold) then 
+      LengthToBreadth = (1.0_r8 + (8.729_r8* &
+        ((1.0_r8 -(exp(-0.03_r8*m_per_min__to__km_per_hour*effective_windspeed)))**2.155_r8)))
```

Comment:
`windspeed_km_hr` should be used here.

Reply:
This is the part that is not bit-for-bit if I change it. Maybe there is a way to set it so that it is b4b?

Reply:
we decided to update this even though we can't make it b4b, diffs are roundoff